stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/channel/logic/ChannelLogic.kt

```diff
@@ -97,6 +100,11 @@ internal class ChannelLogic(
             }
     }
 
+    override suspend fun onChannelMarkReadPrecondition(channelType: String, channelId: String): Result<Unit> =
+        if (ChatClient.instance().needsMarkRead("$channelType:$channelId"))
```

Comment:
It will throw an exception if ChatClient is not initialized but it shouldn't be possible, right?

Reply:
Yes this is invoked when OfflinePlugin is initialized and plugins are initialized when ChatClient is initialized.